sql-plugin/src/main/scala/org/apache/spark/sql/types/rapids/AbstractDataType.scala

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright (c) 2022, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.types.rapids
+
+import org.apache.spark.sql.catalyst.expressions.Expression
+import org.apache.spark.sql.types.{AbstractDataType, AtomicType, DataType, TimestampType}
+
+// Copied from org/apache/spark/sql/types/AbstractDataType.scala
+// for for https://github.com/NVIDIA/spark-rapids/issues/6026
+// It can be removed when Spark 3.3.0 is the least supported Spark version
+private[sql] object AnyTimestampType extends AbstractDataType with Serializable {
```

Comment:
This seems very bad.  We are creating a _separate_ type hierarchy from the one in Apache Spark which we should never do.  Same goes for copying the TimestampNTZType code.  We should _never_ copy the type code.  We should use it directly, or if we cannot because it is not available in all supported Spark versions, shim it in some way as we have done in the past for TimestampNTZType.

Reply:
Added shim code and removed some unnecessary files. But not quite sure the shim source files are under the right packages, please help take another look. Thanks!